src/DevChatter.Bot.Web/Startup.cs

```diff
@@ -32,7 +57,93 @@ public void ConfigureServices(IServiceCollection services)
+
+            services.AddSingleton<HangmanGame>();
+            services.AddSingleton<IBotCommand, HangmanCommand>();
+            services.AddSingleton<IBotCommand, TopCommand>();
+            services.AddSingleton<IBotCommand, UptimeCommand>();
+            services.AddSingleton<IBotCommand, TaxCommand>();
+            services.AddSingleton<IBotCommand, GiveCommand>();
+            services.AddSingleton<IBotCommand, CoinsCommand>();
+            services.AddSingleton<IBotCommand, BonusCommand>();
+            services.AddSingleton<IBotCommand, StreamsCommand>();
+            services.AddSingleton<IBotCommand, ShoutOutCommand>();
+            services.AddSingleton<IBotCommand, QuoteCommand>();
+            services.AddSingleton<IBotCommand, AliasCommand>();
+            services.AddSingleton<IBotCommand, ScheduleCommand>();
+            services.AddSingleton(typeof(IList<>), typeof(List<>));
+            services.AddTransient(typeof(Lazy<>), typeof(Lazier<>));
+
+            services.AddSimpleCommandsFromRepository(repository);
+
+            services.AddSingleton<IBotCommand, HelpCommand>(); // TODO: make these work
+            services.AddSingleton<IBotCommand, CommandsCommand>(); // TODO: make these work
+
+            services.AddSingleton(provider => new CommandList(provider.GetServices<IBotCommand>().ToList()));
+
+            //builder.Register(ctx => new HelpCommand(ctx.Resolve<IRepository>()))
+            //    .OnActivated(e => e.Instance.AllCommands = e.Context.Resolve<CommandList>())
+            //    .AsImplementedInterfaces();
+            //builder.Register(ctx => new CommandsCommand(ctx.Resolve<IRepository>()))
+            //    .OnActivated(e => e.Instance.AllCommands = e.Context.Resolve<CommandList>())
+            //    .AsImplementedInterfaces();
+
+            //builder.Register(ctx => new CommandList(ctx.Resolve<IList<IBotCommand>>()))
+            //    .AsSelf()
+            //    .SingleInstance();
+
+            services.AddSingleton<ICommandUsageTracker, CommandCooldownTracker>();
+            services.AddSingleton<ICommandHandler, CommandHandler>();
+            services.AddSingleton<SubscriberHandler>();
+            services.AddSingleton<IFollowableSystem, FollowableSystem>();
+            services.AddSingleton<IFollowerService, TwitchFollowerService>();
+
+            var api = new TwitchAPI();
+            api.Settings.ClientId = fullConfig.TwitchClientSettings.TwitchClientId;
+            api.Settings.AccessToken = fullConfig.TwitchClientSettings.TwitchChannelOAuth;
+            services.AddSingleton<ITwitchAPI>(api);
+
+            services.AddSingleton<IChatClient, TwitchChatClient>();
+
+            services.AddSingleton<IStreamingInfoService, TwitchStreamingInfoService>();
+
+            services.AddSingleton<BotMain>();
+
+            services.AddSingleton<IHostedService, DevChatterBotBackgroundWorker>();
+
+            services.AddDbContext<AppDataContext>(ServiceLifetime.Transient);
+
```

Comment:
Is this the common way of configuring the services? I find that it's hard to get an overview of what actually happens here. Would it make sense to break up into methods or add comments to mark the different blocks?

Reply:
Yes, we need to have these either get added in through a scanning process or as groups inside of `AddFoo` style extension methods.